pywbem_mock/_mainprovider.py

```diff
@@ -100,25 +105,16 @@ def __init__(self, conn, datastore, default_namespace=DEFAULT_NAMESPACE):
               variables are extracted for the repository including
               the host and disable_pull_operations attributes.
```

Comment:
If we pass the `FakeWBEMConnection` in and store it, this is one of those backlinks that create reference cycles and thus deferred garbage collection. That is not really a problem from a memory perspective since we should not expect very many of those objects, it is more a matter of proper design. It seems that only the attributes `host` and `disable_pull_operations` of the `FakeWBEMConnection` object are actually used. We could pass them in directly.

Reply:
Ahha, you found it.  And it is so easy to do.

This leaves one open limitation for the future.  If, for some reason, a provider wants to use the conn interface to create a client operation back into the providers, that is not possible.  Since we do not discuss it, we can ignore it.

DONE